src/routes/export.ts

```
@@ -0,0 +1,66 @@
+import express from 'express';
+import { Response, Request } from 'express';
+import * as controller from '../controllers/export';
+import { passAsyncError } from '../middleware/error/passAsyncError';
+import { getManager } from 'typeorm';
+import { BadRequestError, NotFoundError } from '../middleware/error/errors';
+import { EnrollmentReport, User } from '../entity';
+import { getSites } from '../controllers/sites';
+
+export const exportRouter = express.Router();
+
+/**
+ * /export/enrollment-report/:reportId GET
+ * Ask the backend to compile a CSV of child objects formatted according
+ * to the data template. Children are found by querying a particular
+ * enrollment report to provide a snapshot of the work the user just
+ * uploaded.
+ */
+exportRouter.get(
+  '/enrollment-report/:reportId',
+  passAsyncError(async (req: Request, res: Response) => {
+    try {
+      const id = parseInt(req.params['reportId']) || 0;
+      const report = await getManager().findOne(EnrollmentReport, id, {
+        relations: [
+          'children',
+          'children.family',
+          'children.family.incomeDeterminations',
+          'children.enrollments',
+          'children.enrollments.site',
+          'children.enrollments.site.organization',
+          'children.enrollments.fundings',
+        ],
+      });
+
+      if (!report) throw new NotFoundError();
+
+      res.send(controller.streamUploadedChildren(res, report.children));
+    } catch (err) {
+      console.error('Unable to download exported enrollment data: ', err);
+      throw new BadRequestError('Could not create spreadsheet to download');
```

Nit: Is this error (and the equivalent in `/export/roster` _really_ indicative of a bad request?  Wondering if 500 error responses might be better suited for catch-all blocks like these.

---

You're probably right, especially because 500 is intended to be more of a catch all block. I'll whip up a 500 error in the middleware and then use that here instead.